core/controllers/acl_decorators_test.py

```diff
@@ -4538,6 +5959,226 @@ def test_any_user_cannot_edit_question(self):
+                expected_status_int=401)
+        error_msg = 'You must be logged in to access this resource.'
+        self.assertEqual(response['error'], error_msg)
+
+    def test_cannot_view_question_editor_with_invalid_question_id(self):
+        invalid_id = 'invalid_question_id'
+        self.login(self.CURRICULUM_ADMIN_EMAIL)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json(
+                '/mock_view_question_editor/%s' % invalid_id,
+                expected_status_int=404)
+        error_msg = (
+            'Could not find the page http://localhost/'
+            'mock_view_question_editor/%s.' % invalid_id
+        )
+        self.assertEqual(response['error'], error_msg)
+        self.logout()
+
+    def test_curriculum_admin_can_view_question_editor(self):
+        self.login(self.CURRICULUM_ADMIN_EMAIL)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json(
+                '/mock_view_question_editor/%s' % self.question_id)
+        self.assertEqual(response['question_id'], self.question_id)
+        self.logout()
+
+    def test_topic_manager_can_view_question_editor(self):
+        self.login(self.user_a_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json(
+                '/mock_view_question_editor/%s' % self.question_id)
+        self.assertEqual(response['question_id'], self.question_id)
+        self.logout()
+
+    def test_normal_user_cannot_view_question_editor(self):
+        self.login(self.user_b_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json(
+                '/mock_view_question_editor/%s' % self.question_id,
+                expected_status_int=401)
+        error_msg = (
+            '%s does not have enough rights to access the questions editor'
+            % self.user_id_b)
+        self.assertEqual(response['error'], error_msg)
+        self.logout()
+
+
+class DeleteQuestionDecoratorTests(test_utils.GenericTestBase):
+    """Tests the decorator can_view_question_editor."""
+
+    question_id = 'question_id'
+    user_a = 'A'
+    user_a_email = 'a@example.com'
+    user_b = 'B'
+    user_b_email = 'b@example.com'
+
+    class MockHandler(base.BaseHandler):
+        GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+        URL_PATH_ARGS_SCHEMAS = {
+            'question_id': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            }
+        }
+        HANDLER_ARGS_SCHEMAS = {'GET': {}}
+
+        @acl_decorators.can_delete_question
+        def get(self, question_id):
+            self.render_json({'question_id': question_id})
+
+    def setUp(self):
+        super(DeleteQuestionDecoratorTests, self).setUp()
+
+        self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
```

Comment:
Do you actually need all the setup here to test the decorator in question (which, AFAICT, just checks the user's role and nothing else)?

Reply:
I have simplified this `setup` and also simplified the `setup` of other tests as well. 
There were many existing tests that also contains some irrelevant/extra code. I have removed those lines.
PTAL!